cmd/agent/subcommands/snmp/command.go

```diff
@@ -302,16 +302,64 @@ func scanDevice(connParams *snmpparse.SNMPConfig, args argsType, snmpScanner snm
 		// newSNMP only returns config errors, so any problem is a usage error
 		return configErr{err}
 	}
-	if err := snmp.Connect(); err != nil {
+	namespace := conf.GetString("network_devices.namespace")
+	deviceID := namespace + ":" + connParams.IPAddress
+	// Since the snmp connection can take a while, start by sending an in progress status for the start of the scan
+	// before connecting to the agent
+	InProgressStatusPayload := metadata.NetworkDevicesMetadata{
```

Comment:
nit: `InProgressStatusPayload`, `ErrorStatusPayload`, `CompletedStatusPayload` don't need to be exported

Reply:
nice catch ! updated in [1cc608e](https://github.com/DataDog/datadog-agent/pull/31894/commits/1cc608e9c88f60a9689fe7e9a3f58bb36a87c936)